TestMissedRevenueHeatMapGenerator.py

```diff
@@ -0,0 +1,187 @@
+                   {'Branch':self.branch, 'Date':self.sale_date_1, 'Increment':-10, 'Sku':self.sku, 'Tag':'sale'},
+                   {'Branch':self.branch, 'Date':self.purchase_date_1, 'Increment':1, 'Sku':self.sku, 'Tag':'purchase'}]
+
+        #TODO: Create a function to calculate amount of hours in a period
+        time_period = (self.time_branch_opens, self.time_branch_closes)
+        amount_of_hours = 8
+        #stock level fell below/equal danger level for 1 hour during this time period
+        danger_period_within_time_period = 1
+        amount_of_sales = 10
+        # average_rate_of_sale_per_hour = 10 sales / 8 hours
+        average_rate_of_sale_per_hour = amount_of_sales / amount_of_hours
+        qty_of_lost_sales = danger_period_within_time_period * average_rate_of_sale_per_hour
+    
+        calculated_qty_of_lost_sales = MR.test_calculate_lost_qty_sales_for_a_period(self.sku, self.branch, journal, time_period, average_rate_of_sale_per_hour)
+        self.assertEqual("calculated_qty_of_lost_sales() returned incorrect value", calculated_qty_of_lost_sales, qty_of_lost_sales)
+
+    def test_calculate_run_rate_for_a_week(self):
+        '''Test the calculated run rate of lost sales for a period.
+           Run rate is used to extrapolate data, in this case amount of sales, to predict quantity of future sales'''
+        #define one week period
+        period = (self.time_branch_opens, self.one_week_later)
+        
+        journal = [{'Branch':self.branch, 'Date':self.sale_date_1, 'Increment':-6, 'Sku':self.sku, 'Tag':'sale'},
+                   {'Branch':self.branch, 'Date':self.sale_date_2, 'Increment':-10, 'Sku':self.sku, 'Tag':'sale'}]
+
+        #calculate average_sales_per_hour for one day: 16 sales / 8 hours
+        average_sales_per_hour = 2
+
+        hours_open_weekday = 8
+        hours_open_weekend = 4
+        #to calculate the weekly run rate, I extrapolate the average sales per hour to a week 
+        expected_weekly_run_rate = average_sales_per_hour * (hours_open_weekday * 5 + hours_open_weekend * 1)
+
+        #get run rate for the defined period
+        calculated_weekly_run_rate = MR.calculate_run_rate(self.sku, self.branch, journal, period)
+        self.assertEqual("calculated_weekly_run_rate() returned incorrect value", calculated_weekly_run_rate, expected_weekly_run_rate)
+    
+    def test_calculate_missed_revenue(self):
+        '''Test calculated missed revenue for a certain sku at a branch
+           Missed revenue equals, missed quantity of sales, multiplied by the retail price of the item'''
+        retail_price = 500
+        qty_of_lost_sales = 3
+        expected_missed_revenue = qty_of_lost_sales * retail_price
+        #method from MissedRevenueHeatmapGenerator
+        calculated_missed_revenue = MR.calculate_missed_revenue(self.sku, self.branch, retail_price, qty_of_lost_sales)
```

Comment:
None of these functions are simply x * y * z where x, y, z are simple numbers. These are f(t), g(t), h(t). Each lost sale and revenue is a function of time, so a single scalar value (e.g. 3) rolls up information.

Model this as a black box: to calc missed revenue, what goes in and what goes out? What is the shape of these data? Are they single numbers? If you think the answer is simply "x * y", you have oversimplified it. Why would a spec contain a function that multiplies x * y? You need to cater for time.